Clean/Circuit/Basic.lean

```diff
@@ -448,336 +448,3 @@ def subcircuit (circuit: FormalCircuit F β α) (b: β.var) := as_stateful (F:=F
-    let goal_add := x + y + carry_in + -1 * z + -1 * (carry_out * 256) = 0
-    show goal_byte ∧ goal_bool ∧ goal_add
-    sorry
-end Add8Full
-
-def add8_wrapped (input : Vector (Expression (F p)) 2) := do
-  let ⟨ [x, y], _ ⟩ := input
-  let z ← subcircuit Add8Full.circuit (vec [x, y, const 0])
-  return z
-
-namespace Add8
-def spec (input : Vector (F p) 2) (z: F p) :=
-  let ⟨ [x, y], _ ⟩ := input
-  z.val = (x.val + y.val) % 256
-
-def assumptions (input : Vector (F p) 2) :=
-  let ⟨ [x, y], _ ⟩ := input
-  x.val < 256 ∧ y.val < 256
-
-def circuit : FormalCircuit (F p) (fields (F p) 2) (field (F p)) where
-  main := add8_wrapped
-  assumptions := assumptions
-  spec := spec
-  soundness := by
-    -- introductions
-    rintro env ⟨ inputs, _ ⟩ ⟨ inputs_var, _ ⟩ h_inputs as
-    let [x, y] := inputs
-    let [x_var, y_var] := inputs_var
-    intro h_holds z
-
-    -- characterize inputs
-    injection h_inputs with h_inputs
-    injection h_inputs with hx h_inputs
-    injection h_inputs with hy h_inputs
-    dsimp at hx hy
-
-    -- simplify constraints hypothesis
-    -- it's just the `subcircuit_soundness` of `Add8Full.circuit`
-    dsimp at h_holds
-
-    -- rewrite input and ouput values
-    rw [hx, hy] at h_holds
-    rw [←(by rfl : z = env 0)] at h_holds
-
-    -- satisfy `Add8Full.assumptions` by using our own assumptions
-    let ⟨ asx, asy ⟩ := as
-    have as': Add8Full.assumptions (vec [x, y, 0]) := ⟨asx, asy, by tauto⟩
-    specialize h_holds as'
-
-    guard_hyp h_holds : Add8Full.circuit.spec (vec [x, y, 0]) z
-
-    -- unfold `Add8Full` statements to show what the hypothesis is in our context
-    dsimp [Add8Full.circuit, Add8Full.spec] at h_holds
-    guard_hyp h_holds : z.val = (x.val + y.val + (0 : F p).val) % 256
-
-    simp at h_holds
-    exact h_holds
-
-  completeness := by
-    -- introductions
-    rintro ⟨ inputs, _ ⟩ ⟨ inputs_var, _ ⟩ h_inputs
-    let [x, y] := inputs
-    let [x_var, y_var] := inputs_var
-    rintro as
-
-    -- characterize inputs
-    injection h_inputs with h_inputs
-    injection h_inputs with hx h_inputs
-    injection h_inputs with hy h_inputs
-    dsimp at hx hy
-
-    -- simplify assumptions and goal
-    dsimp [assumptions] at as
-    dsimp
-    rw [hx, hy]
-
-    -- the goal is just the `subcircuit_completeness` of `Add8Full.circuit`, i.e. the assumptions must hold
-    -- simplify `Add8Full.assumptions` and prove them easily by using our own assumptions
-    dsimp [Add8Full.circuit, Add8Full.assumptions]
-    show x.val < 256 ∧ y.val < 256 ∧ (0 = 0 ∨ 0 = 1)
-    have ⟨ asx, asy ⟩ := as
-    exact ⟨ asx, asy, by tauto ⟩
-
-end Add8
-
-#eval!
-  let p := 1009
-  let p_prime := Fact.mk prime_1009
-  let p_non_zero := Fact.mk (by norm_num : p ≠ 0)
-  let p_large_enough := Fact.mk (by norm_num : p > 512)
-  let main := do
-    let x ← witness (fun _ => 10)
-    let y ← witness (fun _ => 20)
-    add8_wrapped (p:=p) (vec [x, y])
-  main.operations
-end
```

Comment:
maybe we could move that to some test file? during development it was pretty useful to see how the printed constraints changed (or when they couldn't be printed)

Reply:
Yes! I think it got lost into the different files.

Also, we should at some point define a custom command which nicely displays some information about a circuit, ideally without having to resort to concretize a specific prime or specific witness values. This may be non-trivial to do though.